pkg/operator/operator.go

```diff
@@ -225,3 +198,66 @@ func (o *Operator) Start(ctx context.Context) {
 	}()
 	wg.Wait()
 }
+
+func setupIndexers(ctx context.Context, mgr manager.Manager) {
+	err := mgr.GetFieldIndexer().IndexField(ctx, &corev1.Pod{}, "spec.nodeName", func(o client.Object) []string {
+		return []string{o.(*corev1.Pod).Spec.NodeName}
+	})
+	if err != nil {
+		log.FromContext(ctx).Error(err, "failed to setup pod indexer")
```

Comment:
The one question that I have from this: Is this going to cause weird failure states that we don't expect? Like, is it possible that we fail to setup the indexer due to some weird transient error and then we are in this weird half-state that we can't recover from?

We should probably look into the indexer code and see the scenarios where it can return an error. If it's just when the object isn't found, this is probably fine. 

It might also be good to use other projects as reference-points here to see what they do on top of what sig-apimachinery recommends here.